interactor/src/mvx_esdt_safe/mvx_esdt_safe_interactor_main.rs

```diff
@@ -95,30 +94,49 @@ impl MvxEsdtSafeInteract {
     ) {
         let pairs = self.interactor.get_account_storage(&address).await;
 
-        let wanted_hex = hex::encode(wanted_key);
+        let found_entry = pairs.iter().find(|(key, _)| key.contains(wanted_key));
 
-        let found_entry = pairs.iter().find(|(key, _)| key.contains(&wanted_hex));
+        let decoded_key = hex::decode(wanted_key)
+            .ok()
+            .and_then(|bytes| String::from_utf8(bytes).ok())
+            .unwrap_or_else(|| "<invalid utf-8>".to_string());
```

Comment:
If the `wanted_key` cannot be decoded, then `decoded_key` will have value `<invalid utf-8>` or the test will fail?

I would suggest to create a new function for this decoding (and all from below). Also use some assert for decoded to not have value `<invalid utf-8>`